### PR TITLE
Bullseye and Alpine 3.15

### PR DESCRIPTION
**Pull request by J0WI**

### CODE REVIEW

File: update.sh [hunk 1]
```diff
@@ -39,8 +39,8 @@ for version in "${versions[@]}"; do
 
 	fullVersion="$(grep -E "^${rcVersion}([.-]|$)" <<<"$allVersions" | grep $rcGrepV -- "$rcRegex" | head -1)"
 	if [ -z "$fullVersion" ]; then
-		echo >&2 "error: failed to determine latest release for '$version'"
-		exit 1
+		echo >&2 "warning: failed to determine latest release for '$version'"
+		continue
```
Comment (tianon) [inline update.sh:43]: This change seems a little out of place here -- can you elaborate on it?

File: update.sh [hunk 2]
```diff
@@ -82,17 +82,12 @@ for version in "${versions[@]}"; do
...
-		fi
-
 		variantArchCase="$linuxArchCase"
 		if [ "$template" = 'alpine' ]; then
 			sha256="$(grep "julia-${fullVersion}-musl-x86_64.tar.gz$" <<<"$sha256s" | cut -d' ' -f1 || :)"
 			if [ -z "$sha256" ]; then
-				echo >&2 "error: missing musl build for $fullVersion ($version)!"
-				exit 1
+				echo >&2 "warning: missing musl build for $fullVersion ($version)!"
+				continue
```
Comment (tianon) [inline update.sh:90]: Likewise with this change - can you elaborate?

File: update.sh (hunk 1 above)
Comment (J0WI) [reply update.sh:43]: ah yes. The script aborts with `error: missing musl build for 1.0.5 (1.0)!` and `1.6`/`1.7-rc` are not updated anymore.

Comment (tianon) [reply update.sh:43]: I can't reproduce: :grimacing: :confused:

```console
$ ./update.sh
warning: cannot find sha256 for 1.0.5 on arch ppc64le / ppc64le (ppc64el); skipping
1.0: 1.0.5
warning: cannot find sha256 for 1.6.4 on arch ppc64le / ppc64le (ppc64el); skipping
1.6: 1.6.4
warning: cannot find sha256 for 1.7.0-rc3 on arch ppc64le / ppc64le (ppc64el); skipping
1.7-rc: 1.7.0-rc3
```

(All three versions are still listed on https://julialang.org/downloads/, so I don't think they're actually EOL yet, but if they are, we should remove them completely instead.)

Comment (J0WI) [reply update.sh:43]: I can reproduce it in a Debian container (with `bash -x` and `curl` installed):

<details>
<summary>error output</summary>

```bash
+ '[' alpine = debian ']'
+ variantArchCase='dpkgArch="$(dpkg --print-architecture)"; \
	case "${dpkgArch##*-}" in \
# amd64
		amd64) tarArch='\''x86_64'\''; dirArch='\''x64'\''; sha256='\''9dedd613777ba6ebd8aee5796915ff50aa6188ea03ed143cb687fc2aefd76b03'\'' ;; \
# arm32v7
		armhf) tarArch='\''armv7l'\''; dirArch='\''armv7l'\''; sha256='\''cfb2712765db90f0e4fa27e57a88c6d994ebcf1781f8673ebb17b5df7962d0c5'\'' ;; \
# arm64v8
		arm64) tarArch='\''aarch64'\''; dirArch='\''aarch64'\''; sha256='\''41cea1336ed8861413bb945740e567360e26f241eb3e10b3bb0fccd25655ed28'\'' ;; \
# i386
		i386) tarArch='\''i686'\''; dirArch='\''x86'\''; sha256='\''67c8f31699b79df96ce95926a363cd24ffa5bb4d9a814e071b1e8c8ff33e5a8f'\'' ;; \
		*) echo >&2 "error: current architecture ($dpkgArch) does not have a corresponding Julia binary release"; exit 1 ;; \
	esac'
+ '[' alpine = alpine ']'
++ grep 'julia-1.0.5-musl-x86_64.tar.gz$'
++ cut '-d ' -f1
++ :
+ sha256=
+ '[' -z '' ']'
+ echo 'error: missing musl build for 1.0.5 (1.0)!'
error: missing musl build for 1.0.5 (1.0)!
+ exit 1
```
</details>

Comment (tianon) [reply update.sh:43]: Ah, yeah, looks like the `musl` builds of 1.0 are gone -- however, we should fix that by deleting the "alpine" directories in the 1.0 directory (and let the script continue to hard fail for missing things).

Comment (fredrikekre) [reply update.sh]: There never were musl builds for 1.0, Julia 1.5 was the first release with official musl binaries.

**Comment by tianon** [comment]
CI failures are unrelated (something to do with GitHub Actions and Windows 2016 :grimacing:).

Thanks @J0WI!

@fredrikekre thanks for the reminder about `versions.json` :sweat_smile:  it's been on my TODO list to stop scraping the HTML and start using that fabulous resource instead, and I hope to have a PR open soon :smile:

**Pull request merged**